FirebaseVertexAI/Tests/Unit/GenerativeModelTests.swift

```diff
@@ -181,10 +190,13 @@ final class GenerativeModelTests: XCTestCase {
     do {
       _ = try await model.generateContent(testPrompt)
       XCTFail("Should throw GenerateContentError.internalError; no error thrown.")
-    } catch let GenerateContentError.invalidAPIKey(message) {
-      XCTAssertEqual(message, "API key not valid. Please pass a valid API key.")
+    } catch let GenerateContentError.internalError(error as RPCError) {
+      XCTAssertEqual(error.httpResponseCode, 400)
+      XCTAssertEqual(error.status, .invalidArgument)
+      XCTAssertEqual(error.message, "API key not valid. Please pass a valid API key.")
```

Comment:
How can Vertex have an invalid API key? Shouldn't there be a message about a GoogleService-Info.plist?

Reply:
This would only happen if the API key in the `GoogleService-Info.plist` is invalid (e.g., if the key was revoked after the app was shipped). The message/code/status checked here are the ones returned by the backend. I don't think we have specific error cases for revoked/invalid API keys in our other SDKs (everything would be broken in that scenario), which was why I removed the case that was in https://github.com/google/generative-ai-swift/blob/48a0c2f11935a17132492583d4aad51c5a407bcb/Sources/GoogleAI/GenerateContentError.swift#L32-L33 but we could bring it back.